### DISCUSSION TITLE
upgrade path from rook v1.5.3 manifest installation to rook latest helm chart installation

### DISCUSSION THREAD

**Discussion by zkl94** [rook/rook#12305] [Q&A] [answered]
Hi,

Recently we noticed that one of our rook/ceph clusters from two years ago (v1.5.3) is not reclaiming space from PVC. After doing some research, I found out that this space reclaim feature is available in newer version of rook/ceph in the form of csiAddons ReclaimSpaceJob/ReclaimSpaceCronJob. Also, most of our new clusters are already using helm chart based rook/ceph installation (v1.10.12), so I am thinking about upgrade this old cluster from manifest installation to helm chart based installation as well. 

All in all, things I want to achieve and steps I am planning to follow:

1. Upgrade rook/ceph cluster from v1.5.3 to v1.10.12
2. Switch from manifests installation to helm chart based installation
3. Enable csiAddons in the cluster

I have some experience with the 1st item so I am thinking about just following the upgrade paths step by step in https://rook.io/docs/rook/latest/Upgrade/rook-upgrade/ with manifests installation. 

And since csiAddons is available as an option in helm chart, I am planning to enable it after switching to helm chart based installation.

My problem is with the 2nd item: **how to switch from manifest based installation to helm chart based installation**?

And if you see any issues with my upgrade path above, please let me know as well!

**Comment by travisn** [comment, marked as answer]
To start using the helm chart, you could potentially try to add the annotations to the [resources](https://rook.io/docs/rook/latest/Helm-Charts/ceph-cluster-chart/#existing-clusters), then install the helm chart to take over the resource-based install. That topic is only in the cluster chart, but a similar approach is possible for the rook-ceph chart to take over the operator and related resources.

**Reply by zkl94** [reply]
@travisn I follow the guide and successfully take over the operator. But after adding the annotations and labels to the CephCluser CR, I apply rook-ceph-cluster helm chart trying to take over it as well, but the operator decides to delete the CephCluster:

```
NAME        DATADIRHOSTPATH   MONCOUNT   AGE    PHASE      MESSAGE                    HEALTH      EXTERNAL
rook-ceph   /var/lib/rook     3          509d   Deleting   Deleting the CephCluster   HEALTH_OK   false
```

Right now the deletion is blocked because `CephCluster "rook-ceph/rook-ceph" will not be deleted until all dependents
      are removed: CephBlockPool: [ceph-blockpool replicapool], CephFilesystem: [ceph-filesystem],
      CephObjectStore: [ceph-objectstore]`. 

**Could you please suggest a way to revert the deletion?** 

And I really don't know what I did wrong. It should take over instead of deleting it.

**Reply by zkl94** [reply]
By following this guide https://rook.io/docs/rook/latest/Troubleshooting/disaster-recovery/#restoring-crds-after-deletion I am able to bring the cephcluster back. And now both the operator and the cluster are managed by helm charts. I think I have achieved what I wanted! Cheers! @travisn
